Cryptlib/Pk/CryptPkcs7VerifyCommon.c

```diff
@@ -925,6 +932,23 @@ Pkcs7Verify (
   //
   X509_STORE_set_purpose (CertStore, X509_PURPOSE_ANY);
 
+  //
+  // Reject the signature if the signer certificate contains the rejected EKUs
+  // such as ModSign.
+  //
+  EFI_Status = VerifyEKUsInPkcs7Signature(P7Data, P7Length, RejectedEku, 1, FALSE);
+  if (EFI_Status == EFI_SUCCESS) {
+    goto _Exit;

```

Comment:
Should EFI_Status be changed to EFI_SECURITY_VIOLATION?

Reply:
Shouldn't this be `!= EFI_SUCCESS` too?

Reply:
This part checks the rejected EKU, and `Pkcs7Verify()` should return `FALSE` if the rejected EKU exists.
Besides, since `EFI_Status` isn't returned to the caller, we just need to make sure `Status` is still `FALSE`.